core/src/main/scala/org/http4s/headers/Retry-After.scala

```diff
@@ -1,5 +1,26 @@
 package org.http4s
 package headers
 
-object `Retry-After` extends HeaderKey.Default
+import java.time.Instant
+
+import org.http4s.util.{Renderer, Writer}
+import org.http4s.parser.HttpHeaderParser
+
+object `Retry-After` extends HeaderKey.Internal[`Retry-After`] with HeaderKey.Singleton {
+  override def parse(s: String): ParseResult[`Retry-After`] =
+    HttpHeaderParser.RETRY_AFTER(s)
+}
+
+/**
+  * Constructs a `Retry-After` header.
+  *
+  * The value of this field can be either an HTTP-date or an integer number of seconds (in decimal) after the time of the response.
+  *
+  * @param retry Either the date of expiration or seconds until expiration
+  */
+final case class `Retry-After`(retry: Either[Instant, Int]) extends Header.Parsed {
```

Comment:
Yes.  Though I wonder whether `Int` should be `FiniteDuration`, as it represents a time.